docs/src/main/sphinx/admin/fault-tolerant-execution.rst

```diff
@@ -0,0 +1,344 @@
+
+Fault-tolerant execution is a mechanism in Trino that enables a cluster to
+mitigate query failures by retrying queries or their component tasks in
+the event of failure. With fault-tolerant execution enabled, in-progress query
+data is spooled and can be re-used by another worker in the event of a worker
+outage or other fault during query execution.
+
+.. note::
+
+    Fault tolerance does not apply to broken queries or other user error. For
+    example, Trino does not spend resources retrying a query that fails because
+    its SQL cannot be parsed.
+
+Configuration
+-------------
+
+Fault-tolerant execution is disabled by default. To enable the feature, set the
+``retry-policy`` configuration property to either ``QUERY`` or ``TASK``
+depending on the desired :ref:`retry policy <fte-retry-policy>`.
+
+.. code-block:: properties
+
+    retry-policy=QUERY
+
+The following configuration properties control the behavior of fault-tolerant
+execution on a Trino cluster:
+
+.. list-table:: Fault-tolerant execution configuration properties
+   :widths: 30, 40, 30
+   :header-rows: 1
+
+   * - Property name
+     - Description
+     - Default value
+   * - ``retry-policy``
+     - Configures what is retried in the event of failure, either
+       ``QUERY`` to retry the whole query, or ``TASK`` to retry tasks
+       individually if they fail. See :ref:`retry policy <fte-retry-policy>` for
+       more information.
+     - ``NONE``
+   * - ``exchange.deduplication-buffer-size``
+     - Size of the buffer for data sent from worker nodes to the coordinator. If
+       this buffer is filled during query execution, the query fails unless an
+       :ref:`exchange manager <fte-exchange-manager>` is configured.
+     - ``32MB``
+
+.. _fte-retry-policy:
+
+Retry policy
+------------
+
+The ``retry-policy`` configuration property designates whether Trino retries
+entire queries or a query's individual tasks in the event of failure.
+
+QUERY
+^^^^^
+
+A query retry policy instructs Trino to automatically retry a query in the
+event of an error occuring on a worker node. A query retry policy is recommended
+when the majority of the Trino cluster's workload consists of many small
+queries, or if an :ref:`exchange manager <fte-exchange-manager>` is not
+configured.
+
+By default Trino does not implement fault tolerance for queries whose result set
+exceeds 32MB in size, such as :doc:`/sql/select` statements that return a very
+large data set to the user. This limit can be increased by modifying the
+``exchange.deduplication-buffer-size`` configuration property to be greater than
+the default value of ``32MB``, but this will result in higher memory
+utilization on the coordinator.
+
+To enable fault-tolerant execution on queries with a larger result set, it is
+strongly recommended to configure an :ref:`exchange manager
+<fte-exchange-manager>` that utilizes external storage for spooled data and
+therefore allows the in-memory buffer to be kept small.
```

Comment:
For now I've just added it to the `properties-query-management.rst` entry since that's a standard place to put corresponding session properties. Can add it here too if there's consensus.

Reply:
Follow-up, it sounds like the `retry_policy` session property can cause issues for the cluster when the policy is mismatched between sessions. This property was just made to be hidden from the user, so I removed it from this PR as well.